GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -1070,6 +1042,38 @@ - (void)setCurrentUserWithKVO:(GIDGoogleUser *_Nullable)user {
   [self didChangeValueForKey:NSStringFromSelector(@selector(currentUser))];
 }
 
+// Try to retrieve a configuration value from an |NSBundle|'s Info.plist for a given key.
++ (nullable NSString *)configValueFromBundle:(NSBundle *)bundle forKey:(NSString *)key {
+  NSString *value;
+  id configValue = [bundle objectForInfoDictionaryKey:key];
+  if ([configValue isKindOfClass:[NSString class]]) {
+    value = configValue;
+  }
+  return value;
+}
+
+// Try to generate a |GIDConfiguration| from an |NSBundle|'s Info.plist.
++ (nullable GIDConfiguration *)configurationFromBundle:(NSBundle *)bundle {
```

Comment:
Would it be cleaner to put this method (and `+configValueFromBundle:forKey:`) on `GIDConfiguration`?

Reply:
I do think that `configurationFromBundle:` could make sense as a private class method on `GIDConfiguration`.  I'm inclined to wait and try a refactor once we know the outcome of the config conversation with Peter.  The needs may change again.

Reply:
It's up to you. I generally not in favor of keeping this method on `GIDSignIn` whatever the outcome, but it's not an urgent change now.